staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/marshal.go

```diff
@@ -59,6 +60,39 @@ func (s *JSONSchemaPropsOrBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+	return cbor.Marshal(s.Allows)
+}
+
+func (s *JSONSchemaPropsOrBool) UnmarshalCBOR(data []byte) error {
+	if len(data) == 0 {
+		return nil
+	}
+	switch cborType(data[0]) {
+	case cborNullType:
+		return nil
+	case cborBoolType:
+		var b bool
+		if err := cbor.Unmarshal(data, &b); err != nil {
+			return err
+		}
+		*s = JSONSchemaPropsOrBool{Allows: b}
+		return nil
+	case cborMapType:
+		var p JSONSchemaProps
+		if err := cbor.Unmarshal(data, &p); err != nil {
+			return err
+		}
+		*s = JSONSchemaPropsOrBool{Allows: true, Schema: &p}
+		return nil
+	default:
+		return errors.New("boolean or JSON-Schema expected")
```

Comment:
Nit: There's a hyphen in the errors returned by the CBOR implementations that isn't in the JSON implementations.